src/vs/workbench/browser/workbench.ts

```diff
@@ -186,6 +188,11 @@ export class Workbench extends Layout {
 
 		const instantiationService = new InstantiationService(serviceCollection, true);
 
+		// {{SQL CARBON EDIT }}
+		// TODO@Davidshi commandLineService currently has no referents, so force its creation
+		serviceCollection.set(ICommandLineProcessing, instantiationService.createInstance(CommandLineService));
```

Comment:
I think it'd be better to do this via a contribution point. You can see an example of registering a service here :

https://github.com/Microsoft/azuredatastudio/blob/master/src/vs/workbench/services/output/node/outputChannelModelService.ts#L192

And then you load the contribution point like this

See #4675

Reply:
registerSingleton doesn't actual create the service, and how the commandline service is written, it needs to be created on launch in order to work properly.

Reply:
registering it doesn't cause it to be created later? I haven't plumbed into that code yet but I imagine it has to create the services at some point...or are you saying that it'd be created too late for this particular service?

Reply:
It only creates it when it is requests by a class that is being created. CommandLine service isn't required by any class.

Reply:
Makes sense. So theoretically we could add it to something core like the mssql extension (even if it doesn't actually "need" it) and have it loaded?

I'm fine resolving this as is - just want to dig into this for future reference.

Reply:
Arguably it should even be a service since no one is asking for it. This should be built similarly to how vscode builds their command line args. But I believe @shueybubbles avoided this because he wanted to avoid touching vscodes source.

Reply:
we can move this code around when there's sufficient motivation to do so.